cdap-proto/src/main/java/co/cask/cdap/proto/id/ArtifactId.java

```diff
@@ -135,4 +136,16 @@ public static ArtifactId fromIdParts(Iterable<String> idString) {
   public static ArtifactId fromString(String string) {
     return EntityId.fromString(string, ArtifactId.class);
   }
+
+  public co.cask.cdap.api.artifact.ArtifactId toArtifactId() {
+    return new co.cask.cdap.api.artifact.ArtifactId(
+      artifact, new ArtifactVersion(version),
+      NamespaceId.SYSTEM.getEntityName().equals(namespace) ? ArtifactScope.SYSTEM : ArtifactScope.USER);
+  }
+
+  public static ArtifactId from(NamespaceId namespace, co.cask.cdap.api.artifact.ArtifactId id) {
+    return new ArtifactId(ArtifactScope.SYSTEM.equals(
+      id.getScope()) ? NamespaceId.SYSTEM.getNamespace() : namespace.getNamespace(),
```

Comment:
nit: can we get the namespace in a separate line before this? this is a bit hard to read